core/templates/components/common-layout-directives/common-elements/confirm-or-cancel-modal.component.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for simple modal with only two actions: close or
+ * dismiss.
+ */
+
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+
+export class ConfirmOrCancelModalComponent {
+  constructor(protected modalInstance: NgbActiveModal) {}
+
+  confirm(value: unknown): void {
+    this.modalInstance.close(value);
+  }
```

Comment:
Let's make it type-safe. Whenever you are calling this function, you can pass the type. Like this: `this.confirm<TopicData>(topicData)`;
![Screenshot 2021-03-11 at 11 52 40 AM](https://user-images.githubusercontent.com/23002114/110744802-4b7ebb00-8260-11eb-9f56-df4b5986b061.png)

```suggestion
  confirm<T>(value: T): void {
    this.modalInstance.close(value);
  }
```

Reply:
Done